### ISSUE TITLE
AccountBalanceQuery can't handle high request rates in 2.33

### ISSUE THREAD

**Issue by MAkcanca** [hashgraph/hedera-sdk-java#1838]
### Description

Before upgrading to latest version 2.33, there was this interesting approach in the changelog
```
AccountBalanceQuery, AccountInfoQuery, and ContractInfoQuery get part of the data from the Mirror Node REST API (HIP-367)
```
I load tested this approach with Locust, with a simple API that only calls AccountBalanceQuery for the specific account. Previous versions were able to attain 500 Requests per second, meanwhile this hangs and gets congested rather quickly, throwing errors like

```

2024-06-07 13:27:42.767 DEBUG 81 --- [o-8080-exec-192] o.s.web.servlet.DispatcherServlet        : "ERROR" dispatch for GET "/error", parameters={}
2024-06-07 13:27:42.767 ERROR 81 --- [o-8080-exec-198] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.RuntimeException: Error, while processing getAccountTokens mirror node query] with root cause

java.net.ConnectException: HTTP connect timed out
        at java.net.http/jdk.internal.net.http.ResponseTimerEvent.handle(ResponseTimerEvent.java:69) ~[java.net.http:na]
        at java.net.http/jdk.internal.net.http.HttpClientImpl.purgeTimeoutsAndReturnNextDeadline(HttpClientImpl.java:1778) ~[java.net.http:na]
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:1376) ~[java.net.http:na]

2024-06-07 13:27:42.767 DEBUG 81 --- [o-8080-exec-192] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController#error(HttpServletRequest)
2024-06-07 13:27:42.767 DEBUG 81 --- [o-8080-exec-198] o.s.web.servlet.DispatcherServlet        : "ERROR" dispatch for GET "/error", parameters={}
2024-06-07 13:27:42.768 DEBUG 81 --- [o-8080-exec-192] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json, application/json, application/*+json]
2024-06-07 13:27:42.768 DEBUG 81 --- [o-8080-exec-198] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController#error(HttpServletRequest)
2024-06-07 13:27:42.768 DEBUG 81 --- [o-8080-exec-192] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{timestamp=Fri Jun 07 13:27:42 UTC 2024, status=500, error=Internal Server Error, path=/token/balanc (truncated)...]
2024-06-07 13:27:42.768 DEBUG 81 --- [o-8080-exec-198] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json, application/json, application/*+json]
2024-06-07 13:27:42.768 DEBUG 81 --- [o-8080-exec-198] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{timestamp=Fri Jun 07 13:27:42 UTC 2024, status=500, error=Internal Server Error, path=/token/balanc (truncated)...]
```

2.31 does not have this issue, 2.32 did not test.
Reproduced in both testnet and mainnet.

### Steps to reproduce

1. Write a simple Spring boot API that runs AccountBalanceQuery on specified account and start the server.
2. Run Locust with 1000 peak 10 ramp up for the API
3. Wait for results
4. Repeat the steps for v2.33 and v2.31
5. Compare

### Additional context

Locust graph is like below Run 8 (just before Run 9) is 2.31 and Run 9 is 2.33
![image](https://github.com/hashgraph/hedera-sdk-java/assets/9960579/695f54ed-666e-46e7-83ff-95bcfe10efdf)

### Hedera network

testnet

### Version

v.2.33.0

### Operating system

None

**Comment by thenswan** [comment]
Hi @MAkcanca, thanks for submitting this issue. We will review it and get back to you 🙂

**Comment by thenswan** [comment]
Should be closed by #1840
